R/read_csv.R

```diff
@@ -123,6 +123,15 @@ read_cmdstan_csv <- function(files,
   not_matching <- c()
   vroom_warnings <- 0
 
+  no_vroom <- !requireNamespace("vroom", quietly = TRUE)
+  if (no_vroom && !identical(variables, "")) {
+    warning(
```

Comment:
Only throw a warning about missing vroom if reading in variables. This avoids a warning during model fitting because only a few diagnostics are automatically read in. The user will see the warning the first time they access any variables. Other suggestions welcome.

Reply:
This should also check sampler diagnostics are "", I think.

Reply:
I also think if vroom is not present, we should just return everything. I don't think it makes sense to read the entire file in RAM just to then discard some of it and then do it again. Peak RAM usage in this case will be worse than if we warn with something like "no vroom, reading all columns".